src/main/kotlin/org/pkl/lsp/Project.kt

```diff
@@ -54,8 +53,7 @@ class Project(private val server: PklLSPServer) {
 
   val clientCapabilities: PklClientCapabilities by lazy { server.clientCapabilities }
 
-  // The thread where all tree-sitter allocations happen
-  val astExecutor: ExecutorService by lazy { Executors.newSingleThreadExecutor() }
+  val pklParser: PklParser by lazy { PklParser(this) }
```

Comment:
Unrelated: make `PklParser` a component, and remove `astExecutor` because it's only used by the parser anyway.